python/mxnet/gluon/data/dataloader.py

```diff
@@ -87,10 +168,50 @@ def __init__(self, dataset, batch_size=None, shuffle=False, sampler=None,
                              "not be specified if batch_sampler is specified.")
 
         self._batch_sampler = batch_sampler
+        self._num_workers = num_workers
+        if batchify_fn is None:
+            if num_workers > 0:
+                self._batchify_fn = default_mp_batchify_fn
+            else:
+                self._batchify_fn = default_batchify_fn
+        else:
+            self._batchify_fn = batchify_fn
 
     def __iter__(self):
-        for batch in self._batch_sampler:
-            yield _batchify([self._dataset[idx] for idx in batch])
+        if self._num_workers == 0:
+            for batch in self._batch_sampler:
+                yield self._batchify_fn([self._dataset[idx] for idx in batch])
+            return
+
+        key_queue = Queue()
+        data_queue = Queue(2*self._num_workers)
+
+        workers = []
+        for _ in range(self._num_workers):
```

Comment:
Is it possible to reuse shared memory in each iteration? Currently processes are re-created in each iteration, so shared memories are allocated & released a lot of times. On some older kernels (e.g., Linux 3.10 for Centos 7), shm_unlink will not free memory. So the program will OOM quickly when num_workers > 1.

Anyway, this is only related to compatibility. Just ignore it if it is not important.

Reply:
It can be reused. but needs to notify the original process. also memory needs to have the same size every time